Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSReductionCoreEventSlice.py

```diff
@@ -0,0 +1,176 @@
+                                                     optional=PropertyMode.Optional, direction=Direction.Input),
+                             doc='The scatter monitor workspace. This workspace only contains monitors.')
+        # Direct Workspace
+        self.declareProperty(MatrixWorkspaceProperty('DirectWorkspace', '',
+                                                     optional=PropertyMode.Optional, direction=Direction.Input),
+                             doc='The direct workspace.')
+        # Transmission Workspace
+        self.declareProperty(MatrixWorkspaceProperty('TransmissionWorkspace', '',
+                                                     optional=PropertyMode.Optional, direction=Direction.Input),
+                             doc='The transmission workspace')
+
+        self.setPropertyGroup("ScatterWorkspace", 'Data')
+        self.setPropertyGroup("ScatterMonitorWorkspace", 'Data')
+        self.setPropertyGroup("DummyMaskWorkspace", 'Data')
+        self.setPropertyGroup("DirectWorkspace", 'Data')
+        self.setPropertyGroup("TransmissionWorkspace", 'Data')
+
+        # The component
+        allowed_detectors = StringListValidator([DetectorType.to_string(DetectorType.LAB),
+                                                 DetectorType.to_string(DetectorType.HAB)])
+        self.declareProperty("Component", DetectorType.to_string(DetectorType.LAB),
+                             validator=allowed_detectors, direction=Direction.Input,
+                             doc="The component of the instrument which is to be reduced.")
+
+        # The data type
+        allowed_data = StringListValidator([DataType.to_string(DataType.Sample),
+                                            DataType.to_string(DataType.Can)])
+        self.declareProperty("DataType", DataType.to_string(DataType.Sample),
+                             validator=allowed_data, direction=Direction.Input,
+                             doc="The component of the instrument which is to be reduced.")
+
+        # ----------
+        # OUTPUT
+        # ----------
+        # SANSReductionCoreEventSlice has the same outputs as SANSReductionCore
+        self._pyinit_output()
+
+    def PyExec(self):
+        # Get the input
+        state = self._get_state()
+        state_serialized = state.property_manager
+        progress = self._get_progress()
+
+        workspace = self.getProperty("ScatterWorkspace").value
+        # --------------------------------------------------------------------------------------------------------------
+        # 1. Create event slice
+        #    If we are dealing with an event workspace as input, this will cut out a time-based (user-defined) slice.
+        #    In case of a histogram workspace, nothing happens.
+        # --------------------------------------------------------------------------------------------------------------
+        progress.report("Event slicing ...")
```

Comment:
if event slicing only happens on event workspaces, would it be better to move this report so it only occurs if event slicing is actually happening?

Reply:
This algorithm will only be run on event workspaces. I copied over the comment from somewhere else, so I will remove the comment about the input _maybe_ being an event workspace